src/include/dhcp-server/dialogs.rb

```diff
@@ -47,44 +49,58 @@ def ReadDialog
     end
 
     # Write settings dialog
-    # @return `abort if aborted and `next otherwise
+    #
+    # @return [Symbol] :next whether configuration is saved successfully
+    #                  :back if user decided to change settings
+    #                  :abort otherwise
     def WriteDialog
-      Builtins.y2milestone("Running write dialog")
-      Wizard.RestoreHelp(Ops.get(@HELPS, "write", ""))
-      ret = DhcpServer.Write
-      if ret && restart_after_writing?
-        # Restart only if it's already running
-        DhcpServerUI.service.try_restart
-      end
-      # yes-no popup
-      if !ret &&
-          Popup.YesNo(
-            _("Saving the configuration failed. Change the settings?")
-          )
-        return :back
-      end
-      ret ? :next : :abort
+      log.info("Running write dialog")
+
+      help_text = @HELPS.fetch("write") { "" }
+      Wizard.RestoreHelp(help_text)
+
+      return :next if write_settings
+      return :back if Popup.YesNo(_("Saving the configuration failed. Change the settings?"))
```

Comment:
Only for your information: we have a new [Yast2::Popup](https://github.com/yast/yast-yast2/blob/master/library/general/src/lib/yast2/popup.rb) class. Maybe a good opportunity to use it here. You can see an example in [storage-ng](https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2partitioner/widgets/md_devices_selector.rb#L101)

Reply:
Good to know, thank you :+1: